pkg/codegen/go/gen.go

```diff
@@ -3110,8 +3110,10 @@ func generatePackageContextMap(tool string, pkg *schema.Package, goInfo GoPackag
 		pkg.functions = append(pkg.functions, f)
 
 		name := tokenToName(f.Token)
-		originalName := name
-		if pkg.names.Has(name) {
+
+		if pkg.names.Has(name) ||
+			pkg.names.Has(name+"Args") ||
+			pkg.names.Has(name+"Result") {
 			switch {
```

Comment:
I think it would make sense to add a default case to the switch:
```go
default:
    return fmt.Errorf("failed to acquire unique name for %s", name)
```
Right now it looks like this can fail silently.

Reply:
Afraid that might break working programs if conflict detection is inaccurate (i.e. the conflicting type didn't really get generated). Perhaps superstitious.. A warning would be best. I'll look back at this after reInvent.